### PR TITLE
Add test cases for XAConnectionWrapperFactoryTest

### PR DESCRIPTION
**Pull request by complone**
Fixes Issue #17586.

Changes proposed in this pull request:
- Add more test cases for XAConnectionWrapperFactoryTest

### CODE REVIEW

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/DataBaseTypeFixTure.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package org.apache.shardingsphere.transaction.xa.fixture;
```
Comment (yx9o) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/DataBaseTypeFixTure.java:1]: Please add license information in the file header, refer to other files.

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/DataSourceUtils.java [hunk 2]
```diff
@@ -31,13 +31,13 @@
  */
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public final class DataSourceUtils {
-    
+
```
Comment (yx9o) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/DataSourceUtils.java:34]: Please keep the original indentation.

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/DataSourceUtils.java [hunk 3]
```diff
@@ -31,13 +31,13 @@
...
     /**
      * Build data source.
      *
      * @param dataSourceClass data source
-     * @param databaseType database type
-     * @param databaseName database name
+     * @param databaseType    database type
+     * @param databaseName    database name
```
Comment (yx9o) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/DataSourceUtils.java:40]: Please keep the original format.

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/DataBaseTypeFixTure.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+import org.apache.shardingsphere.infra.database.metadata.DataSourceMetaData;
+import org.apache.shardingsphere.infra.database.type.DatabaseType;
+import org.apache.shardingsphere.sql.parser.sql.common.constant.QuoteCharacter;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+
+public class DataBaseTypeFixTure implements DatabaseType {
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/DataBaseTypeFixTure.java:28]: Please fix typo of class name

Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/DataBaseTypeFixTure.java:28]: Please add final if the class is not design for extension

Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/DataBaseTypeFixTure.java:28]: Is it necessary to add this class, how about just use `mock(DatabaseType.class)` ?

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XAConnectionWrapperFixTure.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+import lombok.SneakyThrows;
+import org.apache.shardingsphere.transaction.xa.jta.connection.XAConnectionWrapper;
+
+import javax.sql.XAConnection;
+import javax.sql.XADataSource;
+import java.lang.reflect.Method;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public class XAConnectionWrapperFixTure implements XAConnectionWrapper {
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XAConnectionWrapperFixTure.java:29]: Typo with class name

Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XAConnectionWrapperFixTure.java:29]: Please add final if the class is not design for extension

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XAConnectionWrapperFixTure.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    }
+    
+    @SneakyThrows(ReflectiveOperationException.class)
+    private XAConnection createXAConnection(final XADataSource xaDataSource, final Connection connection) {
+        return (XAConnection) xaConnectionCreatorMethod.invoke(xaDataSource, connection);
+    }
+    
+    @Override
+    public String getType() {
+        return "TEST";
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XAConnectionWrapperFixTure.java:85]: How about rename the type to `FIXTURE`?

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XADataSourceDefinitionFixTure.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package org.apache.shardingsphere.transaction.xa.fixture;
+
+import org.apache.shardingsphere.transaction.xa.jta.datasource.properties.XADataSourceDefinition;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+public class XADataSourceDefinitionFixTure implements XADataSourceDefinition {
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XADataSourceDefinitionFixTure.java:25]: Please add final if the class is not design for extension

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/DataBaseTypeFixTure.java (hunk 4 above)
Comment (terrymanu) [reply shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/DataBaseTypeFixTure.java:28]: The class name is still typo

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XAConnectionWrapperFixTure.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+import lombok.SneakyThrows;
+import org.apache.shardingsphere.transaction.xa.jta.connection.XAConnectionWrapper;
+
+import javax.sql.XAConnection;
+import javax.sql.XADataSource;
+import java.lang.reflect.Method;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public final class XAConnectionWrapperFixTure implements XAConnectionWrapper {
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XAConnectionWrapperFixTure.java:29]: Typo class name

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XADataSourceDefinitionFixTure.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package org.apache.shardingsphere.transaction.xa.fixture;
+
+import org.apache.shardingsphere.transaction.xa.jta.datasource.properties.XADataSourceDefinition;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+public final class XADataSourceDefinitionFixTure implements XADataSourceDefinition {
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XADataSourceDefinitionFixTure.java:25]: Typo class name

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/DataBaseTypeFixture.java [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+import org.apache.shardingsphere.infra.database.metadata.DataSourceMetaData;
+import org.apache.shardingsphere.infra.database.type.DatabaseType;
+import org.apache.shardingsphere.sql.parser.sql.common.constant.QuoteCharacter;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+
+public final class DataBaseTypeFixture implements DatabaseType {
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/DataBaseTypeFixture.java:28]: Database is one word, please do not use `DataBase`

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XAConnectionWrapperDataBaseTypeFixture.java [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+import org.apache.shardingsphere.infra.database.metadata.DataSourceMetaData;
+import org.apache.shardingsphere.infra.database.type.DatabaseType;
+import org.apache.shardingsphere.sql.parser.sql.common.constant.QuoteCharacter;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+
+public final class XAConnectionWrapperDataBaseTypeFixture implements DatabaseType {
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XAConnectionWrapperDataBaseTypeFixture.java:28]: Database is one word, please do not use `DataBase`

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XAConnectionWrapperFixture.java [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+import lombok.SneakyThrows;
+import org.apache.shardingsphere.transaction.xa.jta.connection.XAConnectionWrapper;
+
+import javax.sql.XAConnection;
+import javax.sql.XADataSource;
+import java.lang.reflect.Method;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public final class XAConnectionWrapperFixture implements XAConnectionWrapper {
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/fixture/XAConnectionWrapperFixture.java:29]: Is it necessary to impl such complex fixture?

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/jta/connection/XAConnectionWrapperFactoryTest.java [hunk 13]
```diff
@@ -17,17 +17,46 @@
...
+    public void assertGetInstance() throws SQLException {
+        XAConnectionWrapper xaConnectionWrapperFixTrue = new XAConnectionWrapperFixture();
+        XAConnection actual = xaConnectionWrapperFixTrue.wrap(createXADataSource(), mockConnection());
+        assertThat(actual.getXAResource(), instanceOf(JDBC4MysqlXAConnection.class));
+    }
+    
+    private XADataSource createXADataSource() {
+        DatabaseType databaseType = DatabaseTypeFactory.getInstance("FIXTURE");
+        DataSource dataSource = DataSourceUtils.build(HikariDataSource.class, databaseType, "foo_ds");
+        XADataSourceDefinitionFixture xaDataSourceDefinitionFixTure = new XADataSourceDefinitionFixture();
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/jta/connection/XAConnectionWrapperFactoryTest.java:53]: How about use mock if fixture is not for SPI test?

Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/jta/connection/XAConnectionWrapperFactoryTest.java:53]: Typo variable name `xaDataSourceDefinitionFixTure`

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/jta/connection/XAConnectionWrapperFactoryTest.java (hunk 13 above)
Comment (complone) [reply shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-type/shardingsphere-transaction-xa/shardingsphere-transaction-xa-core/src/test/java/org/apache/shardingsphere/transaction/xa/jta/connection/XAConnectionWrapperFactoryTest.java]: XADataSourceDefinitionFixture needs to get the database connection, the getXADriverClassName interface cannot be called using mock

**Review by terrymanu — APPROVED** [review]

**Pull request merged**